source/elm-live.js

```diff
@@ -52,8 +52,52 @@ module.exports = (argv, options) => {
     return SUCCESS;
   }
 
+  const auxiliaryBuild = (execPath) => {
+    if (!execPath) {
+      return { fatal: false, exitCode: SUCCESS };
+    }
+
+    const process = spawnSync(execPath, [], {
+      stdio: [inputStream, outputStream, outputStream],
+    });
+
+    if (process.error && process.error.code === 'ENOENT') {
+      outputStream.write(
+`\n${dim('elm-live:')}
+  I can’t find the command ${bold(execPath)}!
+  Please make sure you can call ${bold(execPath)}
+  from your command line.
+`
+      );
+
+      return { fatal: true, exitCode: FAILURE };
+    } else if (process.error) {
+      outputStream.write(
+`\n${dim('elm-live:')} Error while calling ${bold(execPath)}! This output may be helpful:
+${indent(String(process.error), 2)}
+
+`
+      );
+    }
+
+    if (args.recover && process.status !== SUCCESS) outputStream.write(
+`\n${dim('elm-live:')}
+  ${bold(execPath)} failed! You can find more info above. Keep calm and take your time
+  to fix your code. We’ll try to compile it again as soon as you change a file.
+
+`
+    );
+
+    return { fatal: false, exitCode: process.status };
+  };
+
   // Build logic
   const build = () => {
+    const beforeBuild = auxiliaryBuild(args.beforeBuild);
+    if (beforeBuild.exitCode !== SUCCESS) {
+      return beforeBuild;
```

Comment:
This one confuses me a bit. build() has about the same behaviour at the bottom of the function. if `args.recover` is true it prints the "something went wrong but you can fix it" message, but it still returns the normal status result.

The only change in behaviour from my point of view is that when `--before-build` fails, `elm-make` is not run. I would argue that this is the correct behaviour because there is a chance `elm-make` would not produce the expected output without the prebuild task running first.